yi-core/src/Yi/CompletionTree.hs

```diff
@@ -27,17 +27,23 @@ module Yi.CompletionTree (
   pretty
   ) where
 
-import           Control.Arrow   (first)
-import           Data.Function   (on)
-import           Data.List       (partition, maximumBy, intercalate)
-import qualified Data.Map.Strict as M
-import           Data.Map.Strict (Map)
-import           Data.Maybe      (isJust, fromJust, listToMaybe, catMaybes)
-import qualified Data.ListLike   as LL
-import           Data.ListLike   (ListLike)
+import           Control.Arrow       (first)
+import           Data.Function       (on)
+import           Data.List           (partition, maximumBy, intercalate)
+import qualified Data.Map.Strict     as M
+import           Data.Map.Strict     (Map)
+import           Data.Maybe          (isJust, fromJust, listToMaybe, catMaybes)
+import qualified Data.ListLike       as LL
+import           Data.ListLike       (ListLike)
+import           Lens.Micro.Platform (over, makeLenses, Lens')
 
 -- | A CompletionTree is a map of partial completions.
-newtype CompletionTree a = CompletionTree (Map a (CompletionTree a)) deriving (Monoid)
+newtype CompletionTree a = CompletionTree {_completionTree :: (Map a (CompletionTree a))}
+  deriving (Monoid)
+
+completionTree :: Lens' (CompletionTree a) (Map a (CompletionTree a))
+completionTree f ct = (\completionTree' -> ct {_completionTree = completionTree'}) <$>
```

Comment:
I did it like this because I dislike TH and it is only one function so it doesn't cause too much clutter.